test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,257 @@
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr invalid with a valid address cookie containing non-zero values")
+    {
+        bm.block->allocsize = 1;
+        bm.block->objectid = 1;
+        bm.block->size = 1024;
+        REQUIRE(test_addr_invalid(s, &bm, 512, 1024, 12345) == 0);
+    }
+
+    SECTION("Test addr invalid with a valid address cookie containing zero values")
+    {
+        bm.block->allocsize = 1;
+        bm.block->objectid = 1;
+        bm.block->size = 0;
+        REQUIRE(test_addr_invalid(s, &bm, 0, 0, 0) == 0);
+    }
+
+    /*
+     * FIXME-WT-13582: Enable once the __wt_panic functions doesn't assert anymore.
+     * SECTION("Test addr invalid address with an invalid address")
+     * {
+     *   bm.block->allocsize = 1;
+     *   bm.block->objectid = WT_TIERED_OBJECTID_NONE;
+     *   bm.block->size = 1024;
+     *   // Create a situation where the block is misplaced, meaning that its address is on the
+     *   // available list.
+     *   utils::off_size_expected test_off = {utils::off_size(512, 4096),
+     *     {
+     *       utils::off_size(512, 4096),
+     *     }};
+     *   REQUIRE(__ut_block_off_insert(s, &bm.block->live.avail, test_off.test_off_size.off,
+     *             test_off.test_off_size.size) == 0);
+     *
+     *
+     *   //Test that the block manager's addr_invalid method returns an error when checking if the
+     *   //address cookie is valid.
+     *   REQUIRE(test_addr_invalid(s, &bm, 512, 1024, 12345) == WT_PANIC);
+     *
+     * }
+     */
+
+    // Cleanup for block created during block manager initialization.
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+    REQUIRE(__wt_block_manager_drop(s, file_path.c_str(), false) == 0);
+}
+
+TEST_CASE("Block manager addr string", "[block_api_misc]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    auto path = std::filesystem::current_path();
+    std::string file_path(path.string() + "/test.wt");
+    setup_bm(session, &bm, file_path);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr string with non-zero values")
+    {
+        // (512, 1024, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 1024, 12345, "[0: 512-1536, 1024, 12345]");
+    }
+
+    SECTION("Test addr string with zero values")
+    {
+        // (0, 0, 0) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 0, 0, 0, "[0: 0-0, 0, 0]");
+    }
+
+    SECTION("Test addr string with zero size")
+    {
+        // (512, 0, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 0, 12345, "[0: 0-0, 0, 0]");
+    }
+
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+}
+
+TEST_CASE("Block header", "[block_api_misc]")
```

Comment:
I prefer using SECTION even if there is only one SECTION in the TEST_CASE.  TEST_CASE states what is being tested.  SECTION states the details of the test.